server/middleware/authenticate.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable */
+/* https://scotch.io/tutorials/authenticate-a-node-js-api-with-json-web-tokens#the-actual-node-application-serverjs */
+import dotenv from 'dotenv';
+import jwt from 'jsonwebtoken';
+dotenv.config()
+const secret = process.env.SECRET_KEY;
+
+export default {
+  
+  checkToken : (req, res) => {
+    // check if there's a token
+    const token = req.body.token || req.params || req.headers['x-access-token'];
+  
+    // decode token
+    if (token) {
+   // console.log(token);
+   // console.log(' token present');
+      // verifies secret 
+      jwt.verify(token, secret, function(err, decoded) {      
+        if (err) {
+          return res.json({ success: false, message: 'Failed to authenticate token.' });    
+        } else {
+            return res.json({ success: 'success', message: 'Access granted' }); 
```

Comment:
Is this function a middleware? if yes; what happens when a user has the correct token?